File: androiddevkit/src/main/java/com/ams/androiddevkit/baseClasses/networking/BaseRestClient.java
package com.ams.androiddevkit.baseClasses.networking;

import java.lang.reflect.ParameterizedType;
import java.util.Objects;

import kotlin.reflect.KClass;

public abstract class BaseRestClient<APIs> extends BaseRetrofitClient {

    public APIs getAPIs() {
        @SuppressWarnings("unchecked")
        Class<APIs> apIsClassType = (Class<APIs>) ((ParameterizedType) Objects.requireNonNull(this.getClass().getGenericSuperclass())).getActualTypeArguments()[0];
        return buildRetrofit().getRetrofitClient(apIsClassType);
    }
}

File: androiddevkit/src/main/java/com/ams/androiddevkit/baseClasses/networking/BaseRestClient.kt
package com.ams.androiddevkit.baseClasses.networking

import com.ams.androiddevkit.baseClasses.networking.interceptor.BaseRefreshTokenInterceptor

abstract class BaseRestClient: RetrofitClient() {

    init {
        buildRetrofit()
    }

    // Seconds.
    override fun getDefaultTimeOut() = 10

    override fun isMockable() = false

    override fun isDebuggable() = true

    override fun getRefreshTokenInterceptor(): BaseRefreshTokenInterceptor<*>? = null
}

